[Edge] Edge worker supports capacity handling instead of concurrency (#43737)
mistercrunch/Airflow: docs/apache-airflow-providers-edge/edge_executor.rst

perspective (you want a worker running from a specific location where required
infrastructure is available).

> Capacity handling

Reviewer: Can you add a note that if not provided default capacity is counted as 1?

Also worthwile to mention that integers are supported, not like K8s where you could reserver `cpu=300millis`

perspective (you want a worker running from a specific location where required
infrastructure is available).

Capacity handling
-----------------

> Some task may need more resources than other tasks, to handle these use case the Edge worker supports
> capacity handling. The logic behind this is the same as the pool slot feature
> see :doc:`apache-airflow:administration-and-deployment/pools`.
> If a task needs more resource, the need_capacity value can be increased. The value can be used to block
> other task from being executed in parallel on the same worker. The need_capacity value works together
> with the capacity value of the worker. A need_capacity of 2 and a worker capacity of 3 means
> that a worker which executes this task can only execute a job with a need_capacity of 1 in parallel.
> If not capacity is defined for a task the default value is 1. The need_capacity value only supports
> integer values.

Reviewer: Some wording nit.
```suggestion
Some tasks may need more resources than other tasks, to handle these use case the Edge worker supports
capacity handling. The logic behind this is the same as the pool slot feature
see :doc:`apache-airflow:administration-and-deployment/pools`.
If a task needs more resources, the ``need_capacity`` value can be increased to reduce concurrency. The value can be used to block
other tasks from being executed in parallel on the same worker. The ``need_capacity`` value works together
with the capacity value of the worker. A ``need_capacity`` of 2 and a worker capacity of 3 means
that a worker which executes this task can only execute a job with a ``need_capacity`` of 1 in parallel.
If no capacity is defined for a task the default value is 1. The ``need_capacity`` value only supports
integer values.
```
